### PR TITLE
Added Rod Cutting [Go]

### LINKED ISSUE #254: Rod Cutting Problem [Go]
#### This issue is about [Rod Cutting Problem](http://www.geeksforgeeks.org/dynamic-programming-set-13-cutting-a-rod/).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.

### PR DESCRIPTION
**Pull request by aashutoshrathi**
Fixes #254 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: rod_cutting_problem/rod_cutting.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package main
+
+import "fmt"
+
+// Computes maximum money that can be earned by cutting a rod of length len(price)
```
Comment (mohitkyadav) [inline rod_cutting_problem/rod_cutting.go]: A function comment looks like
"RodCutting...."

File: rod_cutting_problem/rod_cutting.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import "fmt"
+
+// RodCutting computes maximum money that can be earned by cutting a rod of length len(price)
+// Time Complexity : O((length^2)
+// Space Complexity : O(length)
+func RodCutting(price []int) int {
+	length := len(price)
+	optPrice := make([]int, length+1, length+1)
+
```
Comment (mohitkyadav) [inline rod_cutting_problem/rod_cutting.go]: Do we need a blank line in such a small code?

**Comment by mohitkyadav** [comment]
Do we need a blank line in such a small code.

**Comment by aashutoshrathi** [comment]
Do we ? I'm removing it anyway @mohitkyadav

File: rod_cutting_problem/rod_cutting.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import "fmt"
+
+// RodCutting computes maximum money that can be earned by cutting a rod of length len(price)
+// Time Complexity : O((length^2)
+// Space Complexity : O(length)
+func RodCutting(price []int) int {
+	length := len(price)
+	optPrice := make([]int, length+1, length+1)
```
Comment (mohitkyadav) [inline rod_cutting_problem/rod_cutting.go]: Put space around binary operator `+`.

File: rod_cutting_problem/rod_cutting.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	for i := 1; i <= length; i++ {
+		maxim := -1
+		for j := 0; j < i; j++ {
+			if maxim < price[j]+optPrice[i-j-1] {
```
Comment (mohitkyadav) [inline rod_cutting_problem/rod_cutting.go]: Space around `+`.

**Pull request merged**